packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -0,0 +1,460 @@
+  /// |--------------------------|
+  /// |                          |
+  /// |           Cell           |
+  /// |                          |
+  /// |                          |
+  /// ----------------------------
+  /// ```
+  @override
+  Widget build(BuildContext context) {
+    if (widget.rootNode == null && widget.nodes.isNotEmpty) {
+      return buildSubTreemaps();
+    } else {
+      return buildTreemapCell();
+    }
+  }
+
+  Widget buildTreemapCell() {
+    if (widget.levelsVisible > 0 && widget.rootNode.children.isNotEmpty) {
+      return Padding(
+        padding: const EdgeInsets.all(1.0),
+        child: Column(
+          children: [
+            // TODO(peterdjlee): Abstract out to a widget or a method.
+            Tooltip(
+              message: widget.rootNode.name,
+              child: InkWell(
+                onTap: () {
+                  widget.onRootChangedCallback(widget.rootNode);
+                },
+                child: Container(
+                  width: double.infinity,
+                  decoration: BoxDecoration(border: Border.all(width: 0.5)),
+                  child: buildNameAndSizeText(
+                    fontColor: Colors.white,
+                    onTwoLines: false,
+                  ),
+                ),
+              ),
+            ),
+            Expanded(
+              child: NewTreemap.fromNodes(
+                nodes: widget.rootNode.children,
+                levelsVisible: widget.levelsVisible - 1,
+                onRootChangedCallback: widget.onRootChangedCallback,
+              ),
+            ),
+          ],
+        ),
+      );
+    } else {
+      // TODO(peterdjlee): Abstract out to a widget or a method.
+      return Tooltip(
+        message: widget.rootNode.name,
+        child: InkWell(
+          onTap: () {
+            widget.onRootChangedCallback(widget.rootNode);
+          },
+          child: Container(
+            decoration: BoxDecoration(
+              color: mainUiColor,
+              border: Border.all(width: 0.5),
+            ),
+            child: Center(
+              child: buildNameAndSizeText(
+                fontColor: Colors.black,
+                onTwoLines: true,
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+  }
+
+  Widget buildSubTreemaps() {
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        return Stack(
+          children: buildTreemaps(
+            children: widget.nodes,
+            width: constraints.maxWidth,
+            height: constraints.maxHeight,
+          ),
+        );
+      },
+    );
+  }
+}
+
+class TreemapNode extends TreeNode<TreemapNode> {
+  TreemapNode({
+    @required this.name,
+    this.byteSize = 0,
+    this.childrenMap = const <String, TreemapNode>{},
+  })  : assert(name != null),
+        assert(byteSize != null),
+        assert(childrenMap != null);
+
+  final String name;
+  final Map<String, TreemapNode> childrenMap;
```

Comment:
rename to children

Reply:
It's named childrenMap because children is already defined in the TreeNode class, and we needed to add a separate map field for searching functionalities